lib/pos_fee_estimator.go

```diff
@@ -116,22 +116,36 @@ func (posFeeEstimator *PoSFeeEstimator) AddBlock(block *MsgDeSoBlock) error {
 func (posFeeEstimator *PoSFeeEstimator) addBlockNoLock(block *MsgDeSoBlock) error {
 	// Create a new slice to house the new past blocks and add the new block to it.
 	newPastBlocks := append(posFeeEstimator.cachedBlocks, block)
-	newPastBlocks = posFeeEstimator.cleanUpPastBlocks(newPastBlocks)
+	var removedBlocks []*MsgDeSoBlock
+	newPastBlocks, removedBlocks = posFeeEstimator.cleanUpPastBlocks(newPastBlocks)
 
-	// Create a clean transaction register to add the blocks' transactions.
-	newTransactionRegister := NewTransactionRegister()
-	newTransactionRegister.Init(posFeeEstimator.globalParams.Copy())
+	incomingBlockHash, err := block.Hash()
+	if err != nil {
+		return errors.Wrap(err, "PoSFeeEstimator.addBlockNoLock: error computing blockHash")
+	}
+
+	// Remove all blocks that were pruned are no longer in the cached blocks.
+	for _, removedBlock := range removedBlocks {
+		if err = posFeeEstimator.removeBlockNoLock(removedBlock); err != nil {
```

Comment:
This operation ends up reinitializing the transaction register for every block that's removed. I think it ends up being even less efficient than the original code

Reply:
`removeBlockNoLock` was updated to modify the transaction register in place w/o creating a new one.